internal/federationdomain/endpoints/callback/callback_handler_test.go

```diff
@@ -1227,11 +1262,95 @@ func TestCallbackEndpoint(t *testing.T) {
 			wantDownstreamPKCEChallenge:       downstreamPKCEChallenge,
 			wantDownstreamPKCEChallengeMethod: downstreamPKCEChallengeMethod,
 			wantDownstreamCustomSessionData:   happyDownstreamCustomSessionData,
-			wantAuthcodeExchangeCall: &expectedAuthcodeExchange{
+			wantOIDCAuthcodeExchangeCall: &expectedOIDCAuthcodeExchange{
 				performedByUpstreamName: happyUpstreamIDPName,
 				args:                    happyExchangeAndValidateTokensArgs,
 			},
 		},
+		{
+			name: "GitHub IDP: GET with good state and cookie and successful upstream token exchange returns 303 to downstream client callback",
```

Comment:
Let's add some more GitHub tests to this test table:
- using downstream `response_mode=form_post`
- an error test for when `LoginFromCallback` returns an error for a GitHub IDP
- a GitHub login with identity transforms that change the username and group names
- a GitHub login with identity transforms that reject the auth

Reply:
Done!